app/workers/start_of_cycle_notification_worker.rb

```diff
@@ -0,0 +1,59 @@
+class StartOfCycleNotificationWorker
+  include Sidekiq::Worker
+
+  def perform(service, batch_size = 100)
+    @service = service
+
+    providers_scope.find_each(batch_size: batch_size) do |provider|
+      provider.provider_users.where.not(Arel.sql(chaser_sent_sql)).each do |provider_user|
```

Comment:
do we need to check for chasers here again?

Also separately I wonder if we should start the scope with the provider users with no chaser emails, group by providers, order alphabetically and then slice, or maybe that doesn't work 🤔

Reply:
ok thought about it and that won't work as we might get new users, ignore the second point

Reply:
The `providers_scope` will exclude provider organisations who have users who have received emails (ie. have chaser records) but I think there's a problem with this: If the worker stops/errors mid batch and not all provider users are sent an email we will still omit that provider organisation for the next batch query.
It might be better to check each user for the chaser individually at the cost of additional db work and also persist `ChaserSent` record for the `Provider` when all users have been notified.

We could start the scope with provider users with no chasers and group by providers and order alphabetically but this would mean fetching all users on each worker run as we'd need to order in memory (that `find_each` ordering limitation). That might not be such a big deal given the numbers and it will be an ever decreasing resultset due to the filtering of users who have 'ChaserSent` records. There's also the problem of the batch size passed to `find_each` when we group by provider as this will be variable depending on the total number of users in the organisations in the batch. So that seems like a difficult approach but I might be missing an easier way to achieve this.